rule-examples/src/main/java/com/solibri/smc/api/examples/beginner/UIComponentExample.java

```diff
@@ -167,12 +186,75 @@ public UIContainer getParametersUIDefinition() {
 		 */
 		mainContainer.addComponent(createContainerWithLabelAndRadioButtonWithImages());
 
+		mainContainer.addComponent(createCustomStringParameterComponent());
 		/*
 		 * Return the container that contains all of the UI components for the parameters panel.
 		 */
 		return mainContainer;
 	}
 
+	/**
+	 * A custom UI for a string parameter that presents a text panel and a button.
+	 * Modifications to the text panel are committed when the button is pressed.
+	 */
+	private final class CustomStringParameterComponent implements RuleParameterCustomUi {
+
+		private final StringParameter customStringParameter;
+		private final JTextPane textPane = new JTextPane();
+		private final CommitAction commitAction = new CommitAction();
+
+		@SuppressFBWarnings(value = "SE_BAD_FIELD", justification = "This class is not serialized")
+		private class CommitAction extends AbstractAction {
+			@Serial private static final long serialVersionUID = -1;
+
+			CommitAction() {
+				putValue(Action.NAME, resources.getString("CommitButton.NAME"));
+			}
```

Comment:
Could add an empty row after this constructor.

Reply:
⬆️